### PR TITLE
Initial commit of bash-completion 2.1

### PR DESCRIPTION
**Pull request by alarcher**
This is an initial addition of the bash-completion macros maintained by Debian as I need this to handle the environment-modules in a clean way.
- Some macros are not useful on OpenIndiana and other might need modifications but at least this commit would give the possibility for people to get started with.
- A variable BASH_COMPLETIONS_PATH has been added to shared-macros.mk to allow installation of macros by other packages in the same directory.

As it is my first addition to oi-userland I would appreciate your comments and advice on the best practices.
Best regards,

Aurelien

### CODE REVIEW

File: components/bash-completion/bash-completion.p5m [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+<transform file path=.*\.py$ -> default pkg.tmp.autopyc false>
+
+dir  path=etc
+dir  path=etc/profile.d
+file path=etc/profile.d/bash_completion.sh
```
Comment (pyhalov) [inline components/bash-completion/bash-completion.p5m]: Here should be preserve attribute and 0644 file mode

Comment (alarcher) [inline components/bash-completion/bash-completion.p5m]: Thank you, is there a page listing attributes that should be usually enforced in the manifest ?
On another topic, should I force a dependency to bash or is it generated automatically ?

Comment (pyhalov) [inline components/bash-completion/bash-completion.p5m]: I don't know about such page. But you could look at other manifests. I think that dependency on bnash will not be generated automaticaly - such  dependency is usually  based on requiring particular shared libraries. And this package doesn't provide anything for pkgdepend to find out that bash is required. So, yes, manual depend action seems to be required.

**Comment by xen0l** [comment]
Hi,

and what about existing /etc/bash/bash_completion.example from bash? I guess it would be best to move this as well to the package if possible.

**Comment by alarcher** [comment]
Hi,
I added the attributes for bash_completion.sh and an explicit dependency to shell/bash.
I am not sure what to do with /etc/bash/bash_completion.example because it might be redundant with the completions present in /ust/share/bash-completion/completions.
Best regards,

Aurelien

**Comment by pyhalov** [comment]
Pehaps, we can just remove this file from bash manifest?

**Comment by xen0l** [comment]
Let's do it then. But please do so in a separate commit.

**Comment by alarcher** [comment]
I cleaned up the commit history. I will remove the bash_completion.example file in another pull request.

File: components/bash-completion/bash-completion.p5m [hunk 2]
```diff
@@ -0,0 +1,608 @@
...
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+<transform file path=.*\.py$ -> default pkg.tmp.autopyc false>
+
+depend type=require fmri=pkg:/shell/bash
+
+dir  path=etc
+dir  path=etc/profile.d
```
Comment (pyhalov) [inline components/bash-completion/bash-completion.p5m:29]: the same note - please, firstly add group=sys rule for profile.d and profile.d/\* to transforms/defaults

**Comment by pyhalov** [comment]
please, modify pull request so that it can be automatically merged

**Comment by alarcher** [comment]
Should be OK now.

**Pull request merged**